docs/functions/building-functions.md

```diff
@@ -0,0 +1,42 @@
+# Building functions
+
+--8<-- "build-func-intro.md"
+
+## Local builds
+
+You can build a container image for your function locally without deploying it to a cluster, by using the `build` command.
+
+### Prerequisites
+
+- You have a Docker daemon on your local machine. This is already provided if you have used the Quickstart installation.
+
+### Procedure
+
+--8<-- "proc-building-function.md"
+
+## On-cluster Builds
+
+If you do not have a local Docker daemon running, or you are using a CI/CD pipeline, you might want to build your function on the cluster instead of using a local build. You can create an on-cluster build by using the `func deploy --remote` command.
+
+### Prerequisites
+
+- The function must exist in a Git repository.
+- You must configure your cluster to use Tekton Pipelines. See the [on-cluster build](https://github.com/knative/func/blob/main/docs/reference/on_cluster_build.md){target=_blank} documentation.
+
+### Procedure
+
+When running the command for the first time, you must specify the Git URL for the function:
+
+=== "func"
+
+    ```{ .console }
+    func deploy --remote --registry <registry> --git-url <git-url> -p hello
+    ```
+
+=== "kn func"
+
+    ```{ .console }
+    kn func deploy --remote --registry <registry> --git-url <git-url> -p hello
+    ```
+
+After you have specified the Git URL for your function once, you can omit it in subsequent commands.
```

Comment:
This is the tricky part of the direct-upload change that was recently made. If you have specified the Git URL in a previous command, and then subsequently run `func deploy --remote` as you describe here, the source will be pulled from the git repo. If you later want to perform a direct upload deployment, you need to run `func deploy --remote --git-url=""` which is pretty terrible, and probably a point in favor of not yet documenting the direct-upload method until we get the UX improved.

Reply:
Can you open a new issue maybe for this so that we can merge it as is and update it later? This is all based on what was provided in the other PR, so technically making changes at this point we're starting to add scope creep to what was originally meant to be published in the docs for kubecon this week, and it's going to end up delaying the whole PR.